File: DSA/akashyadav_590014126/Day_41_Question_3.py
def remove_k_digits(num,k):
    stack=[]

    for digit in num:
        while stack and k>0 and stack[-1] > digit:
            stack.pop()
            k -= 1
        stack.append(digit)

    final_stack=stack[:-k] if k else stack

    result = ''.join(final_stack).lstrip('0')

    return result if result else '0'

#example
print(remove_k_digits("1432219",3))

File: DSA/upama_kumari_590014909/day_41_question_3.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char* removeKdigits(char* num, int k) {
    int n = strlen(num);
    char* stack = (char*)malloc((n + 1) * sizeof(char));
    int top = -1;

    for (int i = 0; i < n; i++) {
        while (top >= 0 && stack[top] > num[i] && k > 0) {
            top--;
            k--;
        }
        stack[++top] = num[i]; 
    }

    top -= k;
    if (top < -1) top = -1;

    stack[top + 1] = '\0'; 

    int i = 0;
    while (stack[i] == '0' && i <= top) i++;

    char* result = strdup(stack + i);
    if (strlen(result) == 0)
        return strdup("0");

    return result;
}

int main() {
    char num[] = "10200";
    int k = 1;
    char* smallest = removeKdigits(num, k);
    printf("Output: %s\n", smallest); 
    free(smallest);
    return 0;
}
